### PR TITLE
Add RHBK configuration

### PR DESCRIPTION
**Pull request by mdujava**

### CODE REVIEW

**Review by azgabur — APPROVED** [review]
`no-ssl-sso` gets created successfully following steps for Kuadrant testsuite tools deploy

File: base/rhbk/deploy-rhsso.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  done
+}
+
+function deployRHSSO {
+  DB_USERNAME="${DB_USERNAME}" DB_PASSWORD="${DB_PASSWORD}" envsubst < "${FILE_ROOT}"/db-credentials.yaml.tpl | oc apply -n "${NAMESPACE}" -f -
+
+  TARGET_NAMESPACE="${NAMESPACE}" envsubst < "${FILE_ROOT}"/operator-group.yaml.tpl | oc apply -n "${NAMESPACE}" -f -
+  oc apply -n "${NAMESPACE}" -f "${FILE_ROOT}"/keycloak-subscription.yaml
+
+  waitSuccess oc wait installplan --all --for=condition=Installed -n "${NAMESPACE}"
```
[the remark this answers is not in the document]
Comment (mdujava) [reply base/rhbk/deploy-rhsso.sh:36]: Install plan is not created right away, so we need to wait on creation and then on the completion.

**Review by JaurbanRH — DISMISSED** [review]

File: base/rhsso/deploy-rhsso.sh [hunk 2]
```diff
@@ -4,38 +4,26 @@ set -exuo pipefail
...
-  cat ${FILE_ROOT}/credential-sso-secret.yaml.tpl | ADMIN_USERNAME=${ADMIN_USERNAME} ADMIN_PASSWORD=${ADMIN_PASSWORD} envsubst | oc apply -f - -n ${NAMESPACE}
-  oc apply -f "${FILE_ROOT}"/sso-keycloak.yaml -n ${NAMESPACE}
-  oc apply -f "${FILE_ROOT}"/no-ssl-sso-service.yaml -n ${NAMESPACE}
-  oc apply -f "${FILE_ROOT}"/no-ssl-sso-route.yaml -n ${NAMESPACE}
-  waitSuccess oc rollout status statefulset/keycloak -w -n ${NAMESPACE}
-
-  oc --namespace ${NAMESPACE} rsh statefulset/keycloak bash -c "/opt/eap/bin/kcadm.sh update realms/master -s sslRequired=NONE --server http://no-ssl-sso:8080/auth --realm master --user ${ADMIN_USERNAME} --password ${ADMIN_PASSWORD} --no-config"
+  TARGET_NAMESPACE="${NAMESPACE}" envsubst < "${FILE_ROOT}"/operator-group.yaml.tpl | oc apply -f - -n "${NAMESPACE}"
+  oc apply -f "${FILE_ROOT}"/keycloak-subscription.yaml -n "${NAMESPACE}"
+  oc wait -n "${NAMESPACE}" --for=jsonpath=status.installPlanRef.name subscription rhsso-operator --timeout=120s
```
Comment (averevki) [inline base/rhsso/deploy-rhsso.sh:16]: Is this working for you? I just tried to run this command without any resources and: `jsonpath wait format must be --for=jsonpath='{.status.readyReplicas}'=3`. I would suggest to wait on `status.installPlanRef.kind=InstallPlan` if this error is real

File: base/rhsso/deploy-rhsso.sh [hunk 3]
```diff
@@ -4,38 +4,26 @@ set -exuo pipefail
...
+  oc wait -n "${NAMESPACE}" --for=condition=Installed installplan --all --timeout=120s
+
+  ADMIN_USERNAME="${ADMIN_USERNAME}" ADMIN_PASSWORD="${ADMIN_PASSWORD}" envsubst < "${FILE_ROOT}"/credential-sso-secret.yaml.tpl | oc apply -f - -n "${NAMESPACE}"
+  oc apply -f "${FILE_ROOT}"/sso-keycloak.yaml -n "${NAMESPACE}"
+  oc apply -f "${FILE_ROOT}"/no-ssl-sso-service.yaml -n "${NAMESPACE}"
+  oc apply -f "${FILE_ROOT}"/no-ssl-sso-route.yaml -n "${NAMESPACE}"
+  timeout 120 bash -c "oc get statefulset -w -n "${NAMESPACE}" -o name | grep -qm1 keycloak"
```
Comment (averevki) [inline base/rhsso/deploy-rhsso.sh:23]: Nice solution :+1:

File: base/rhsso/deploy-rhsso.sh (hunk 2 above)
Comment (mdujava) [reply base/rhsso/deploy-rhsso.sh:16]: Looks like its new format introduced in version 4.15.0

Comment (averevki) [reply base/rhsso/deploy-rhsso.sh:16]: You right, time to have it installed from AUR :grimacing:

**Review by averevki — DISMISSED** [review]

**Review by azgabur — DISMISSED** [review]
`no-ssl-sso` gets created successfully following steps for Kuadrant testsuite tools deploy

**Comment by azgabur** [comment]
Not sure what changed but on Openshift 4.15 the `no-ssl-sso` does not work because of error at the end of script:

```
+ oc rsh -n tools statefulset/keycloak bash -c '/opt/eap/bin/kcadm.sh update realms/master -s sslRequired=NONE --server http://no-ssl-sso:8080/auth --realm master --user admin --password admin --no-config'
Defaulted container "keycloak" out of: keycloak, extensions-init (init)
Logging into http://no-ssl-sso:8080/auth as user admin of realm master
HTTPS required [invalid_request]
command terminated with exit code 1
```

**Review by averevki — APPROVED** [review]
LGTM

**Review by averevki — APPROVED** [review]

**Review by JaurbanRH — APPROVED** [review]

**Review by azgabur — APPROVED** [review]
`no-ssl-sso` gets created successfully following steps for Kuadrant testsuite tools deploy

**Pull request merged**